app/src/main/java/org/thoughtcrime/securesms/loki/protocol/MultiDeviceProtocol.kt

```diff
@@ -76,12 +81,18 @@ object MultiDeviceProtocol {
     // TODO: remove this after we migrate to new message receiving pipeline
     @JvmStatic
     fun handleConfigurationMessage(context: Context, content: SignalServiceProtos.Content, senderPublicKey: String, timestamp: Long) {
-        if (TextSecurePreferences.getConfigurationMessageSynced(context)) return
+        if (TextSecurePreferences.getConfigurationMessageSynced(context) && !TextSecurePreferences.shouldUpdateProfile(context, timestamp)) return
```

Comment:
We might want to have separate timestamps for the display name and the profile picture

Reply:
The reason being that a user might update their display name and then their profile picture (two separate events), but you might get the profile picture update first and then ignore the display name one because it's from before the last profile update :)

Reply:
that's true, might even be good to check if and only update the profile pic if it HAS changed by comparing them before setting

Reply:
with your case above though shouldn't the later update still have the more up to date profile name in the out of order received messages?